buildSrc/src/main/groovy/com.google.api-ads.java-conventions.gradle

```diff
@@ -58,67 +63,165 @@ dependencies {
 
-  public ExampleRunnerTask() {
+  ExampleRunnerTask() {
     group = 'Execution'
     description = 'Run a Google Ads API example.'
     errorOutput = System.err
   }
 
+  String getBasePackage() {
+    return this.basePackage
+  }
+
+  void setBasePackage(String basePackage) {
+    this.basePackage = basePackage
+  }
+
+  String getExampleArguments() {
+    return this.exampleArguments
+  }
+
   @Option(option = 'example', description = 'Sets the example to launch and' +
-      ' any arguments. Required for execution. E.g. ' +
-      '"basicoperations.GetCampaigns --customerId 1234567890"')
-  public void setExampleArguments(String exampleArguments) {
+    ' any arguments. Required for execution. E.g. ' +
+    '"basicoperations.GetCampaigns --customerId 1234567890"')
+  void setExampleArguments(String exampleArguments) {
     this.exampleArguments = exampleArguments.trim()
     int firstSpaceIndex = this.exampleArguments.indexOf(' ')
 
     // No additional arguments were passed, just the example name.
     if (firstSpaceIndex == -1) {
-      main = basePackage + this.exampleArguments
+      main = this.basePackage + this.exampleArguments
     }
     // Otherwise, separate the input and set the arguments to pass to the
     // main class.
     else {
-      main = basePackage + this.exampleArguments[0..firstSpaceIndex - 1]
+      main = this.basePackage + this.exampleArguments[0..firstSpaceIndex - 1]
       argsString(exampleArguments[(firstSpaceIndex + 1)..-1])
     }
   }
 
   @TaskAction
   @Override
-  public void exec() {
+  void exec() {
     if (!(exampleArguments?.trim())) {
       throw new GradleException('\033[0;31mMissing example!\033[0m ' +
-          'Please rerun with one provided, e.g. ' +
-          '\033[0;35m--example="basicoperations.GetCampaigns"\033[0m')
+        'Please rerun with one provided, e.g. ' +
+        '\033[0;35m--example="basicoperations.GetCampaigns"\033[0m')
     }
     logQuietMessage('Running example: ' + main + ', args: ' +
-        args.toString())
+      args.toString())
     try {
       super.exec()
     }
     catch (Exception e) {
       logQuietMessage('\n\033[0;31mrunExample exception!\033[0m Did ' +
-          'you provide a valid example identifier? E.g. ' +
-          '\033[0;35m--example="basicoperations.GetCampaigns"\033[0m\n\n' +
-          e.message)
+        'you provide a valid example identifier? E.g. ' +
+        '\033[0;35m--example="basicoperations.GetCampaigns"\033[0m\n\n' +
+        e.message)
     }
   }
 
-  public String getExampleArguments() {
-    return exampleArguments
-  }
-
   private void logQuietMessage(String message) {
     logger.quiet(message)
   }
+}
+
+class ListExamplesTask extends DefaultTask {
+
+  // The base directory containing all examples.
+  private final String examplesDirectory =
+    'google-ads-examples/src/main/java/com/google/ads/googleads/examples/'
+
+  // Regex for extracting descriptions.
+  @Input
+  private final String regex =
```

Comment:
Can be static.

Reply:
Fixed; great point, this simplifies things a bit.